NightscoutServiceKit/OTPManager.swift

```diff
@@ -0,0 +1,223 @@
+//
+//  OTPManager.swift
+//  Loop
+//
+//  Created by Jose Paredes on 3/28/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import LoopKit
+import OneTimePassword
+import Base32
+
+private let OTPSecretKeyService = "OTPSecretKeyService"
+private let OTPSecretKeyCreatedService = "OTPSecretKeyCreatedService"
+private let OTPRecentAcceptedPasswordsService = "OTPRecentAcceptedPasswordsService"
+
+public protocol OTPSecretStore {
+    
+    func tokenSecretKey() -> String?
+    func setTokenSecretKey(_ key: String?) throws
+    
+    func tokenSecretKeyName() -> String?
+    func setTokenSecretKeyName(_ name: String?) throws
+    
+    func recentAcceptedPasswords() -> [String]
+    func setRecentAcceptedPasswords(_ passwords: [String]) throws
+}
+
+extension KeychainManager: OTPSecretStore {
+    
+    public func tokenSecretKey() -> String? {
+        return try? getGenericPasswordForService(OTPSecretKeyService)
+    }
+    
+    public func setTokenSecretKey(_ key: String?) throws {
+        try replaceGenericPassword(key, forService: OTPSecretKeyService)
+    }
+    
+    public func tokenSecretKeyName() -> String? {
+        return try? getGenericPasswordForService(OTPSecretKeyCreatedService)
+    }
+    
+    public func setTokenSecretKeyName(_ name: String?) throws {
+        try replaceGenericPassword(name, forService: OTPSecretKeyCreatedService)
+    }
+    
+    public func recentAcceptedPasswords() -> [String] {
+        guard let recentString = try? getGenericPasswordForService(OTPRecentAcceptedPasswordsService) else {
+            return []
+        }
+
+        return convertRecentAcceptedPasswordsFromString(recentString)
+    }
+    
+    public func setRecentAcceptedPasswords(_ passwords: [String]) throws {
+        try replaceGenericPassword(convertRecentAcceptedPasswordsToString(passwords), forService: OTPRecentAcceptedPasswordsService)
+    }
+    
+    func convertRecentAcceptedPasswordsToString(_ recentAcceptedPasswords: [String]) -> String {
+        return recentAcceptedPasswords.joined(separator: ",")
+    }
+    
+    func convertRecentAcceptedPasswordsFromString(_ passwordsString: String) -> [String] {
+        return passwordsString.split(separator: ",").map({String($0)})
+    }
+}
+
```

Comment:
Manages overall OTP operations, using the store above (keychain) to track OTP secret and the last 2 OTP codes.